[pranavyey] iP

pranavyey opened the pull request:
### yeyAI

> "The true art of memory is the art of attention." - Samuel Johnson 

yeyAI helps you remember things you need to do. It's,

- text-based
- easy to learn
- **SUPER FAST** to use
- All you need to do is,

1. download it from [here](https://github.com/pranavyey/ip/releases/tag/A-Jar).
2. double-click it.
3. add your tasks.

let it manage your tasks for you 😉
And it is FREE!

Features:

- [x]  Managing tasks
- [ ]  Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public static void main(String[] args) {
        new YeyAI(System.getProperty("user.dir") + "/tasks.txt").run();
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

pranavyey committed:
rename main class and add greeting

pranavyey committed:
Level-1 Echo

pranavyey committed:
Level-2

pranavyey committed:
Level-3

pranavyey committed:
Level-4

pranavyey committed:
Level-5

pranavyey committed:
Level-5-fix

pranavyey committed:
Level-6

pranavyey committed:
Level-6 bugs

pranavyey committed:
A-TextUitesting

pranavyey committed:
Added tasks.txt file to .gitignore

pranavyey committed:
Level-7

pranavyey committed:
Level-7 branch merge

pranavyey committed:
Level 7 Done boolean bugfix

pranavyey committed:
Level 7 done boolean bugfix

Benjam11n commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,16 @@
+public class Deadline extends Task{
```
Do ensure that there is a space between Task and { in the method signature

Benjam11n commented on src/main/java/yeyAI.java:
```diff
@@ -0,0 +1,153 @@
+        if (Files.exists(path)) {
+            try {
+                List<String> lines = Files.readAllLines(path);
+                for (String taskLine : lines){
```
Do also ensure that there is a space between for (String taskLine : lines) and { in the for loop.

Benjam11n commented on src/main/java/yeyAI.java:
```diff
@@ -0,0 +1,153 @@
+        System.out.println(line + "Hello! I'm yeyAI\n" + "What can I do for you?\n" + line);
+        String input = scanner.nextLine();
+        while (!input.equals("bye")) {
+            try {
```
Would it be better to place `String input = scanner.nextLine();` within the while loop? This would eliminate the need for repeated input = scanner.nextLine(); statements in each branch and prevent potential infinite loops if someone forgets to read the next input.

Benjam11n commented on src/main/java/yeyAI.java:
```diff
@@ -0,0 +1,153 @@
+                    tasks.get(taskIndex).setDone();
+                    System.out.println("Nice! I've marked this task as done:");
+                    System.out.println(tasks.get(taskIndex).toString());
+                    input = scanner.nextLine();
```
I noticed that for the deadline branch, you checked for whether the task the user intended to delete was actually in the list. Could similar checks for the unmark and mark commands be implemented?

```
if (Integer.parseInt(parts[1]) > tasks.size()) {
    throw new YeyException("Task to be deleted must exist within list");
}
```

Benjam11n reviewed (commented):
Overall good work so far!

pranavyey committed:
retrieve stored deadlines and events bugfix

pranavyey committed:
Level-8

pranavyey committed:
Merge Level-8 date feature

pranavyey committed:
More-OOP

pranavyey committed:
A-Packages

pranavyey committed:
Merge branch 'add-gradle-support'

pranavyey committed:
A-Gradle

pranavyey committed:
A-JUnit

pranavyey committed:
Added JavaDoc comments

pranavyey committed:
Merge pull request #1 from pranavyey/branch-A-JavaDoc

Merge branch-A-JavaDoc to master

pranavyey committed:
Ensured coding standard

pranavyey committed:
Merge branch 'branch-A-CodingStandard'

pranavyey committed:
find matching tasks

pranavyey committed:
Merge branch 'branch-Level-9'

pranavyey committed:
Added JavaDoc comments for Level-9

pranavyey committed:
Added GUI functionality

pranavyey committed:
Add assertions for key components in yeyAI

Assertions were added to TaskList, Storage and Parser to ensure
that critical assumptions hold true during execution.This helps
catch potential errors early and improves code reliability.

Assertions include:
- Ensuring valid task indices before modifying the list
- Verifying that loaded tasks have correct formatting
- Checking that event start dates are before end dates

This commit improves the robustness of the application by preventing
invalid states from occurring.

pranavyey committed:
Merge pull request #3 from pranavyey/branch-A-Assertions

Add assertions for TaskList, Storage, and Parser

pranavyey committed:
Refactor code to improve readability and maintainability

Several methods in the Parser and Task classes contained large blocks
of logic that made the code difficult to read and maintain. The
processCommand() function did not fulfill CodeQuality requirements
previously.

This commit introduces:
- Extraction of logic in 'processCommand()' into seperate helper
  methods ('markTask()', 'deleteTask()' etc.)
- Improved error handling for invalid task formats and index validation.
- Added assertions to enforce critical assumptions, such as ensuring
  valid indices.

pranavyey committed:
Merge branch 'master' into branch-A-CodeQuality

pranavyey committed:
Merge pull request #4 from pranavyey/branch-A-CodeQuality

Refactor code to improve readability and maintainability

pranavyey committed:
Add edit functionality

pranavyey committed:
Prevent updating event dates incorrectly

pranavyey committed:
Add user guide
